src/models/transaction.ts

```diff
@@ -0,0 +1,348 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { CREATE_TOKEN_TX_VERSION, DEFAULT_TX_VERSION, DECIMAL_PLACES, TOKEN_INFO_VERSION, TX_WEIGHT_CONSTANTS, MAX_INPUTS, MAX_OUTPUTS } from '../constants';
+import { crypto, encoding, util } from 'bitcore-lib';
+import helpers from '../utils/helpers';
+import Input from './input';
+import Output from './output';
+import { CreateTokenTxInvalid, MaximumNumberInputsError, MaximumNumberOutputsError } from '../errors';
+import buffer from 'buffer';
+
+type optionsType = {
+  version?: number,
+  weight?: number,
+  nonce?: number,
+  timestamp?: number | null,
+  parents?: string[],
+  tokens?: string[],
+  hash?: string | null,
+  name?: string | null,
+  symbol?: string | null,
+};
+
+const defaultOptions: optionsType = {
+  version: DEFAULT_TX_VERSION,
+  weight: 0,
+  nonce: 0,
+  timestamp: null,
```

Comment:
Maybe it should be the current timestamp?

Reply:
I think it should be null here. We can initialize in the constructor or create a `setCurrentTimestamp()` method.

Reply:
What's the difference between initializing here or in the constructor? I think it would be ok for the default value to be the current timestamp, just like the version. I would just need to take care to execute the timestamp in the moment the constructor is created, not when I set this object. If it gets too complicated I can set in the constructor.

Reply:
This code is executed when the module is loaded. The constructor is executed each time an object is created.

Reply:
Yeah, makes sense. I will keep it null for now and we can change later if we think it's better.